### PR TITLE
Adding coverage task configuration

### PR DESCRIPTION
**Pull request by mattinger**

### CODE REVIEW

**Comment by mattinger** [comment]
The intent here is to add the task to do coverage verification, but not necessarily tie it to the check lifecycle.  That may be a decision best left up to the build file author on whether or not they want it tied to the lifecycle, or to call it explicitly.

**Comment by alexeisele** [comment]
Any timeline for the release of this change by chance?
Would love to incorporate it into an application I'm working on

File: src/test/groovy/com/vanniktech/android/junit/jacoco/GenerationTest.groovy [hunk 1]
```diff
@@ -376,7 +379,46 @@ class GenerationTest {
...
+          assert additionalSourceDirs.contains(project.file("src/debug/$it"))
+        }
+
+        assert sourceDirectories.size() == 10
+        LANGUAGES.every {
+          assert sourceDirectories.contains(project.file("src/main/$it"))
+          assert sourceDirectories.contains(project.file("src/debug/$it"))
+        }
+
+        // TODO: Validate Rules
```
Comment (vanniktech) [inline src/test/groovy/com/vanniktech/android/junit/jacoco/GenerationTest.groovy:400]: Is this something you want to do?

File: README.md [hunk 2]
```diff
@@ -9,6 +9,12 @@ Gradle plugin that generates Jacoco reports from a Gradle Project. Android Appli
   - Executes the `test<Flavor><BuildType>UnitTest` task before
   - Gets executed when the `check` task is executed
   - Generated Jacoco reports can be found under `build/reports/jacoco/<Flavor>/<BuildType>`.
+  
+- Task `jacocoTestCoverageVerification<Flavor><BuildType>`
+  - Depends on the `jacocoTestReport<Flavor><BuildType>` task
+  - Verifies that your code coverage limits have not been violated
+  - Must be manually specified in your gradle targets to execute
```
Comment (vanniktech) [inline README.md:16]: How about?

```suggestion
  - Must be manually invoked
```

File: README.md [hunk 3]
```diff
@@ -39,6 +45,11 @@ When having `debug` & `release` build types and `red` & `blue` flavors the follo
   - Executes the `test` task before
   - Gets executed when the `check` task is executed
   - Generated Jacoco reports can be found under `build/reports/jacoco/`.
+  
+- Task `jacocoTestCoverageVerification`
+  - Depends on the `jacocoTestReport` task
+  - Verifies that your code coverage limits have not been violated
+  - Must be manually specified in your gradle targets to execute
```
Comment (vanniktech) [inline README.md:52]:
```suggestion
  - Must be manually invoked
```

File: README.md [hunk 4]
```diff
@@ -93,6 +104,18 @@ junitJacoco {
   excludes // type String List
   includeNoLocationClasses = false // type boolean
   includeInstrumentationCoverageInMergedReport = false // type boolean
+  violationRules {
+    failOnViolation = true
+    rule {
+      enabled = true
+      element = "BUNDLE"
```
Comment (vanniktech) [inline README.md:111]: What's this one for?

**Review by vanniktech — APPROVED** [review]
Sorry, lost track of this. Have you been able to test it out? @alexeisele

**Comment by alexeisele** [comment]
@vanniktech thanks for following up!

Have not been able to test as my project is behind a corporate proxy, however if I have time this weekend I will try on a separate test protect.

**Comment by athkalia** [comment]
@mattinger has there been any update on this?

**Comment by geft** [comment]
@alexeisele @mattinger Any update?

**Comment by vanniktech** [comment]
@mattinger want to resolve the conflicts?

**Comment by vanniktech** [comment]
I'm closing this issue due to inactivity. If you have any further input on the issue, don't hesitate to reopen this issue or post a new one.

**Pull request closed without merging**